packages/@aws-cdk/core/lib/private/refs.ts

```diff
@@ -201,7 +201,8 @@ function getCreateExportsScope(stack: Stack) {
 }
 
 function generateExportName(stackExports: Construct, id: string) {
-  const stackRelativeExports = stackExports.node.tryGetContext(cxapi.STACK_RELATIVE_EXPORTS_CONTEXT);
+  const stackRelativeExports = stackExports.node.tryGetContext(cxapi.STACK_RELATIVE_EXPORTS_CONTEXT) ??
```

Comment:
Since feature flags are always in the context (at least when they are used in the construct library), we can remove the boilerplate:

```ts
const stackRelativeExports = FeatureFlags.isEnabled(this, cxapi.STACK_RELATIVE_EXPORTS_CONTEXT);
```